### PR TITLE
Implement JSON API manually to enable schema introspection for docs

### PR DESCRIPTION
**Pull request by glowka**

### CODE REVIEW

**Comment by glowka** [comment]
OK, so making a lot of little steps I ended up ...with massive reimplementation, but without changes to url endpoints and general logic of the views. Also, a lot of dedicated tooling used, like: inspectors, renders, parsers, filters, but only in limited scope and only because they perfectly matched (witch is not always the case) the purpose of retaining views and serializers simple.

**Comment by glowka** [comment]
In the end I went all the way and also cleaned imports etc, which is probably not the best practice to do it all in one PR, but changes were already massive, so I would treat this merge like a release or something of that size.

File: settings/base.py [hunk 1]
```diff
@@ -163,9 +163,13 @@
 
 SWAGGER_SETTINGS = {
     'DEFAULT_FIELD_INSPECTORS': [
+        'apps.pp.serializers.RootSerializerInspector',
         'apps.pp.serializers.IDFieldInspector',
         'drf_yasg.inspectors.CamelCaseJSONFilter',
-        'drf_yasg.inspectors.ReferencingSerializerInspector',
+        # ReferencingS... replaced with InlineS... which does not create serializers definitions index,
+        # but does not require serializers class names to be unique across whole application
+        # 'drf_yasg.inspectors.ReferencingSerializerInspector',
```
Comment (kaglowka) [inline settings/base.py]: Nice comment :+1:

File: apps/pp/serializers.py [hunk 2]
```diff
@@ -1,86 +1,160 @@
...
+            model = Reference
+            fields = ('url', 'ranges', 'quote',
+                      'priority', 'comment', 'reference_link', 'reference_link_title')
+
+    class Relationships(serializers.Serializer):
+        class ReferenceRequest(ResourceSerializer):
+            pass
+
+        reference_request = data_wrapped(ReferenceRequest(required=False, allow_null=True))(required=False,
+                                                                                            allow_null=True)
```
Comment (kaglowka) [inline apps/pp/serializers.py]: Is this double `(required=False, allow_null=True)` correct?

Comment (glowka) [reply apps/pp/serializers.py]: Yup, it was correct. If we have `{reference_request : { data: {.. }}` first pair is for outer key and the other for the nested/

I changed a bit interface to make call a bit more intuitive https://github.com/PrzypisPowszechny/pp/pull/8, also somehow related tests for set_relationship.

**Pull request merged**